File: DP/2096-going_down.cpp
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int N;
    cin >> N;

    vector<int> maxDp(3);
    vector<int> minDp(3);
    cin >> maxDp[0] >> maxDp[1] >> maxDp[2];
    minDp = maxDp;

    for (int i = 1; i < N; i++)
    {
        int p1, p2, p3;
        cin >> p1 >> p2 >> p3;

        maxDp = {p1 + max(maxDp[0], maxDp[1]), p2 + max(maxDp[0], max(maxDp[1], maxDp[2])), p3 + max(maxDp[1], maxDp[2])};
        minDp = {p1 + min(minDp[0], minDp[1]), p2 + min(minDp[0], min(minDp[1], minDp[2])), p3 + min(minDp[1], minDp[2])};
    }

    cout << max(maxDp[0], max(maxDp[1], maxDp[2])) << " " << min(minDp[0], min(minDp[1], minDp[2])) << endl;
}

File: DP/2096-going_down.py
import sys

board = []

N = int(sys.stdin.readline())

max_sum = [0 for i in range(3)]
min_sum = [0 for i in range(3)]

max_sum = list(map(int, sys.stdin.readline().split(' ')))
min_sum = max_sum[:]


for _ in range(1, N):
    a, b, c = map(int, sys.stdin.readline().split())
    
    max_sum = [a + max(max_sum[0], max_sum[1]), b + max(max_sum), c + max(max_sum[1], max_sum[2])]
    min_sum = [a + min(min_sum[0], min_sum[1]), b + min(min_sum), c + min(min_sum[1], min_sum[2])]

print(max(max_sum), min(min_sum))